node/src/neighborhood/mod.rs

```diff
@@ -2253,40 +2284,73 @@ mod tests {
     }
 
     #[test]
-    fn test_wtih_standard_gossip_have_new_exit_node_in_database() {
+    fn init_db_countries_works_properly() {
         let mut subject = make_standard_subject();
-        let root_node_key = subject.neighborhood_database.root_key().clone();
-        let root_node = subject.neighborhood_database.root().clone();
-        let first_neighbor = make_node_record(1111, true);
-        let second_neighbor = make_node_record(2222, true);
-
+        subject.min_hops = Hops::OneHop;
+        let root_node_key = subject.neighborhood_database.root().public_key().clone();
+        let mut first_neighbor = make_node_record(1111, true);
+        first_neighbor.inner.country_code_opt = Some("CZ".to_string());
+        let mut second_neighbor = make_node_record(2222, true);
+        second_neighbor.inner.country_code_opt = Some("DE".to_string());
         subject
             .neighborhood_database
             .add_node(first_neighbor.clone())
             .unwrap();
+        subject
+            .neighborhood_database
+            .add_node(second_neighbor.clone())
+            .unwrap();
         subject
             .neighborhood_database
             .add_arbitrary_full_neighbor(&root_node_key, first_neighbor.public_key());
-        let mut gossip_db = subject.neighborhood_database.clone();
-
-        gossip_db.set_root_key(first_neighbor.public_key());
-        gossip_db.remove_node(&root_node_key);
-        gossip_db.add_node(root_node).unwrap();
-        gossip_db.add_arbitrary_full_neighbor(first_neighbor.public_key(), &root_node_key);
-        gossip_db.add_node(second_neighbor.clone()).unwrap();
-        gossip_db
-            .add_arbitrary_full_neighbor(first_neighbor.public_key(), second_neighbor.public_key());
-        gossip_db.root_mut().inner.version = 1;
-        let resigner = gossip_db
-            .node_by_key_mut(first_neighbor.public_key())
+        subject
+            .neighborhood_database
+            .add_arbitrary_full_neighbor(&root_node_key, second_neighbor.public_key());
+        let filled_db_countries = subject.init_db_countries();
+
+        subject
+            .neighborhood_database
+            .remove_arbitrary_half_neighbor(&root_node_key, first_neighbor.public_key());
+        subject
+            .neighborhood_database
+            .remove_arbitrary_half_neighbor(&root_node_key, second_neighbor.public_key());
+        let emptied_db_countries = subject.init_db_countries();
+
+        assert_eq!(filled_db_countries, &["CZ".to_string(), "DE".to_string()]);
+        assert!(emptied_db_countries.is_empty());
```

Comment:
Vic prukazny by bylo, kdybys odebral pouze jeden. Tim vyloucis, ze to neni tak, ze kdyz zkusis odeberat jeden, vzdy skoncics s uplne resetovanou databazi. ;)